### PR TITLE
STYLE: fix pylints redefined-outer-name warnings

### PR DESCRIPTION
**Pull request by bhaveshrp**
Fixed pylint redefined-outer-name warnings for following 2 files

**1. pandas\core\frame.py
2. pandas\core\dtypes\inference.py**

- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/dtypes/inference.py [hunk 1]
```diff
@@ -417,8 +417,7 @@ def is_dataclass(item):
 
     """
     try:
-        from dataclasses import is_dataclass
-
-        return is_dataclass(item) and not isinstance(item, type)
+        import dataclasses as dcs
+        return dcs.is_dataclass(item) and not isinstance(item, type)
```
Comment (MarcoGorelli) [inline pandas/core/dtypes/inference.py:421]: can we just `import dataclasses` and then return
```
dataclasses.is_dataclass(item) and not isinstance(item, type)
```

File: pandas/core/frame.py [hunk 2]
```diff
@@ -6493,9 +6493,9 @@ def drop_duplicates(
 
         inplace = validate_bool_kwarg(inplace, "inplace")
         ignore_index = validate_bool_kwarg(ignore_index, "ignore_index")
-        duplicated = self.duplicated(subset, keep=keep)
+        duplicate = self.duplicated(subset, keep=keep)
 
-        result = self[-duplicated]
+        result = self[-duplicate]
```
Comment (MarcoGorelli) [inline pandas/core/frame.py:6498]: let's just inline this one
```
result = self[-self.duplicated(subset, keep=keep)]
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for working on this

File: pandas/core/dtypes/inference.py (hunk 1 above)
Comment (bhaveshrp) [reply pandas/core/dtypes/inference.py]: Sure. This is perfect. I am working on this

File: pandas/core/frame.py (hunk 2 above)
Comment (bhaveshrp) [reply pandas/core/frame.py:6498]: Sure. Thanks!

**Comment by MarcoGorelli** [comment]
should be fine thanks, I'll approve / merge on green

**Review by MarcoGorelli — APPROVED** [review]
thanks @bhaveshrp

**Pull request merged**